### DISCUSSION TITLE
Multiple dynamic key definition for a model

### DISCUSSION THREAD

**Discussion by ghost** [pydantic/pydantic#3305] [Question]
I'm trying to achieve the below JSON is there a way to use multiple dynamic keys in models
For e.g.
```
{
    "Stages":{
        "id": 1,
        "stage{id}":{
            "a":"a"
        },
        "process{id}":{
            "b":"b"
        }
    }
}

```
How would I define a schema for something like this, also the child schema for process and stage is fixed but their key is dynamic?

so I created a Stages schema, Stage schema, and Process schema where id can be defined but what about other n number of process and stage schemas.

Also, I was thinking if it's possible to have pattern-based alias something similar to this [3303](https://github.com/samuelcolvin/pydantic/discussions/3303)

**Comment by eode** [comment]
voluptuous.
